lib/Raven/ErrorHandler.php

```diff
@@ -155,10 +158,13 @@ public function registerExceptionHandler($call_existing_exception_handler = true
         $this->call_existing_exception_handler = $call_existing_exception_handler;
     }
 
-    public function registerErrorHandler($call_existing_error_handler = true, $error_types = -1)
+    public function registerErrorHandler($call_existing_error_handler = true, $error_types = null)
     {
+        if ($error_types == null) {
+            $error_type = error_reporting();
+        }
         $this->error_types = $error_types;
-        $this->old_error_handler = set_error_handler(array($this, 'handleError'), error_reporting());
+        $this->old_error_handler = set_error_handler(array($this, 'handleError'), -1);
```

Comment:
we're basically changing the logic to say "always call our error handler, we'll determine later if we care about the event"